src/libcprover-cpp/options.h

```diff
@@ -20,6 +26,10 @@ class api_optionst
 
   api_optionst &simplify(bool on);
 
+  api_optionst &drop_unused_functions(bool on);
+
+  api_optionst &validate_goto_model(bool on);
+
```

Comment:
It looks like there is some duplication between the options and the session here. Am I reading this correctly? The passes are specified as enabled or not in the options, but they are only acted on when the calls to api_session is made, which ignores the values of these options.

Reply:
Yes, you read it right - this is underspecified there and left as an example.

We want to see tomorrow what's the best way to act on the options - one thought is that the options should act at the side of the `api_sessiont` (calling the corresponding functions if the option is enabled, but do it silently).

We will see tomorrow, this is still being worked on (if anything, it needs to integrate the changes from your `api_messaget` introducing PR), but there needs to be some thought. This is just a check-in for the end of the day as a PR.